src/solidity-frontend/pattern_check.cpp

```diff
@@ -0,0 +1,119 @@
+#include <solidity-frontend/pattern_check.h>
+
+pattern_checker::pattern_checker(const nlohmann::json &_ast_nodes,
+    const std::string &_target_func, const messaget &msg):
+    ast_nodes(_ast_nodes),
+    target_func(_target_func),
+    msg(msg)
+{
+}
+
+bool pattern_checker::pattern_checker::do_pattern_check()
+{
+  // add more functions here to perform more pattern-based checks
+  //print_json_element(ast_nodes);
+  printf("Checking function %s ...\n", target_func.c_str());
```

Comment:
Why don't we use our messaging interface based on the fmt library?

Reply:
Done.